packages/stk/stk_util/stk_util/ngp/NgpSpaces.hpp

```diff
@@ -39,57 +39,15 @@
 namespace stk {
 namespace ngp {
 
-#ifdef KOKKOS_ENABLE_CUDA
-using ExecSpace = Kokkos::Cuda;
-#elif defined(KOKKOS_ENABLE_HIP)
-using ExecSpace = Kokkos::Experimental::HIP;
-#elif defined(KOKKOS_ENABLE_OPENMP)
-using ExecSpace = Kokkos::OpenMP;
-#else
-using ExecSpace = Kokkos::Serial;
-#endif
+using ExecSpace = Kokkos::DefaultExecutionSpace;
 
-#ifdef KOKKOS_ENABLE_CUDA
-using HostExecSpace = Kokkos::Serial;
-#elif defined(KOKKOS_ENABLE_HIP)
-using HostExecSpace = Kokkos::Serial;
-#elif defined(KOKKOS_ENABLE_OPENMP)
-using HostExecSpace = Kokkos::OpenMP;
-#else
-using HostExecSpace = Kokkos::Serial;
-#endif
+using HostExecSpace = Kokkos::DefaultHostExecutionSpace;
 
-#ifdef KOKKOS_ENABLE_CUDA
-using MemSpace = Kokkos::CudaSpace;
-#elif defined(KOKKOS_ENABLE_HIP)
-using MemSpace = Kokkos::Experimental::HIPSpace;
-#elif defined(KOKKOS_ENABLE_OPENMP)
-using MemSpace = Kokkos::OpenMP;
-#else
-using MemSpace = Kokkos::HostSpace;
-#endif
+using MemSpace = ExecSpace::memory_space;
 
-#ifdef KOKKOS_ENABLE_CUDA
-#ifdef KOKKOS_ENABLE_CUDA_UVM
-using UVMMemSpace = Kokkos::CudaUVMSpace;
-#else
-using UVMMemSpace = Kokkos::CudaHostPinnedSpace;
-#endif
-#elif defined(KOKKOS_ENABLE_HIP)
-using UVMMemSpace = Kokkos::Experimental::HIPHostPinnedSpace;
-#elif defined(KOKKOS_ENABLE_OPENMP)
-using UVMMemSpace = Kokkos::OpenMP;
-#else
-using UVMMemSpace = Kokkos::HostSpace;
-#endif
+using UVMMemSpace = Kokkos::SharedSpace;
```

Comment:
In the case of KOKKOS_ENABLE_HIP, is Kokkos::SharedSpace the same as HIPHostPinnedSpace ?
I think that needs to be true. Otherwise this needs to be tested on HIP before we accept these changes. @ldh4 what do you think?

Reply:
No, `Kokkos::SharedSpace` is `Kokkos::HIPManagedSpace` for `HIP` but that wasn't available initially and my suspicion is that it was just not updated.